src/Mod/Assembly/App/AssemblyObject.h

```diff
@@ -26,10 +26,8 @@
 #define ASSEMBLY_AssemblyObject_H
 
 
-#include <GeomAbs_CurveType.hxx>
-#include <GeomAbs_SurfaceType.hxx>
-
 #include <Mod/Assembly/AssemblyGlobal.h>
+#include <Mod/Assembly/App/AssemblyUtils.h>
```

Comment:
The include should probably be in the .cpp file no? I guess you did this to have access to JointType here perhaps? If so you can do a forward declaration here and still put the header in the cpp.